Add DropZone component

jmophoto opened the pull request:
Adds drop zone component that will accept a file, either by dragging or through a file browser, and return that file.

Props:
- onFileAdd: function returning the file dropped or selected
- onFileRemove: function called on remove click
- uploadedFile: file that is passed back to the component on drop/select

This incorporates @tumentumurchudur's hover effect, changing the border to dashed blue and a white background when dragging onto the window.

Validation and file processing is left up to the parent component. 

DropZone name could use some suggestions. ImageUpload doesn't seem to work because any file can be dropped. FileUpload might work, but the component doesn't actually upload. Thoughts @mxenabled/frontend-engineers?

Initial State:
<img width="929" alt="screen shot 2016-03-02 at 10 05 50 pm" src="https://cloud.githubusercontent.com/assets/488916/13485460/34f1c888-e0c5-11e5-8dc1-56c6e52092bc.png">

Drag State:
<img width="927" alt="screen shot 2016-03-02 at 10 06 35 pm" src="https://cloud.githubusercontent.com/assets/488916/13485459/34f1bd02-e0c5-11e5-9836-3d7b6ff25416.png">

Loaded State:
<img width="919" alt="screen shot 2016-03-02 at 10 06 50 pm" src="https://cloud.githubusercontent.com/assets/488916/13485461/34f2343a-e0c5-11e5-9172-dc86486c7330.png">

bsbeeks commented:
For the name, I vote `FileUpload`.

bsbeeks commented on src/components/DropZone.js:
```diff
@@ -0,0 +1,165 @@
+        imageSource: reader.result
+      });
+      this.props.onFileAdd(file);
+    }.bind(this);
```
instead of binding, you can use a fat arrow method.

bsbeeks commented on src/components/DropZone.js:
```diff
@@ -0,0 +1,165 @@
+  }
+});
+
+const styles = {
```
let's move this into a method called `styles` to follow our new internal pattern.

bsbeeks commented:
Overall, this is a solid. I think @tumentumurchudur 's component has some elements that yours is missing (https://github.com/mxenabled/mx-react-components/pull/188). Specifically I think the validation (file type, size, etc) should be included in the component. I suggest combining the PRs into one.

jmophoto commented:
This has been cleaned up and combined with items from @tumentumurchudur's PR.

- Added validation for type and size (optional)
- Added validation message on error
- Validation message is removed on new drag
- Drag indicator was changed from window to the drop zone
- On drag
-- Expand height of the drop zone
-- Change background to white
-- Add dashed blue border
-- Add import icon
-- Change text
- Add conditional to show preview if image and document icon for others
- Renamed to FileUpload

@mxenabled/frontend-engineers

bsbeeks commented on src/components/FileUpload.js:
```diff
@@ -0,0 +1,220 @@
+
+const FileUpload = React.createClass({
+  propTypes: {
+    allowedfileTypes: React.PropTypes.array,
```
should be `allowedFileTypes`...the "f" needs to be capitalized.

jmophoto replied:
:+1: Fixed and pushed.

bsbeeks commented:
:shipit:
